src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -2079,6 +2079,14 @@ public void observe(@NonNull SCMHead head, @NonNull SCMRevision revision) throws
                     } else if (revision.isDeterministic()) {
                         SCMRevision scmLastBuiltRevision = _factory.getRevision(project);
                         SCMRevision scmLastSeenRevision = _factory.getLastSeenRevision(project);
+                        if (scmLastSeenRevision == null && scmLastBuiltRevision != null) {
+                            scmLastSeenRevision = scmLastBuiltRevision;
+                            try {
+                                _factory.setLastSeenRevisionHash(project, revision);
+                            } catch (IOException e) {
+                                printStackTrace(e, listener.error("Could not update last seen revision hash"));
+                            }
+                        }
                         if (!revision.equals(scmLastSeenRevision)) {
```

Comment:
The above is good upgrade protection so should probably be there in any case.
Though as @fbelzunc suggests, to keep the previous version's behaviour the if statement should be changed.
```suggestion
                        if (!revision.equals(scmLastBuiltRevision)) {
```

Reply:
Agreed